source/Octopus.Client/Repositories/Async/MixedScopeBaseRepository.cs

```diff
@@ -41,16 +42,18 @@ void ValidateThatICanUseACustomSpaceContext()
             }
         }
 
-        protected override void CheckSpaceResource(IHaveSpaceResource spaceResource)
+        protected override Task CheckSpaceResource(IHaveSpaceResource spaceResource)
         {
-            Repository.Scope.Apply(
+            return Repository.Scope.Apply(
```

Comment:
Actually this can be simplified by returning the task at the end

```
protected override Task CheckSpaceResource(IHaveSpaceResource spaceResource)
{
    Repository.Scope.Apply(
        whenSpaceScoped: space =>
                {
                    if (spaceResource.SpaceId != null && spaceResource.SpaceId != space.Id)
                        throw new ResourceSpaceDoesNotMatchRepositorySpaceException(spaceResource, space);
                },
                whenSystemScoped: () => { },
                whenUnspecifiedScope: () => { });
    return Task.FromResult(null);
}
```